src/block_service/fds/fds_service.cpp

```diff
@@ -674,12 +673,10 @@ dsn::task_ptr fds_file_object::download(const download_request &req,
     std::shared_ptr<std::ofstream> handle(new std::ofstream(
         req.output_local_name, std::ios::binary | std::ios::out | std::ios::trunc));
     if (!handle->is_open()) {
-        char buffer[512];
-        char *ptr = strerror_r(errno, buffer, 512);
-        derror("fds download failed: fail to open localfile(%s) when download(%s), error(%s)",
-               req.output_local_name.c_str(),
-               _fds_path.c_str(),
-               ptr);
+        derror_f("fds download failed: fail to open localfile({}) when download({}), error({})",
```

Comment:
I found that there are a log code refactor in this pull request. So in my opinion,  it's better to split it into server small pull requests.

Reply:
In fact this is also a fix to adapt to macOS, using the portable function `safe_strerror` instead of `strerror_r`.